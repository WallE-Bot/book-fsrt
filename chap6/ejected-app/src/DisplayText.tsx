import React, { useState } from 'react';

const DisplayText = () => {
  const [txt, setTxt] = useState('');
  const [msg, setMsg] = useState('');

  const onChangeTxt = (e: React.ChangeEvent<HTMLInputElement>) => {
    setTxt(e.target.value);
  }

  const onClickShowMsg = (
      e: React.MouseEvent<HTMLButtonElement, MouseEvent>
    ) => {
      e.preventDefault();

      setMsg(`Welcome to React testing, ${txt}`);
    }

  return (
    <form>
      <label>Enter your name</label>
      <input
        data-testid='user-input'
        value={txt}
        onChange={onChangeTxt}
      />
      <button
        data-testid='input-submit'
        onClick={onClickShowMsg}
      >
        Show Message
      </button>
      <label
        data-testid='final-msg'
      >
        {msg}
      </label>
      <p>test entry</p>
    </form>
  );
}

export default DisplayText;
