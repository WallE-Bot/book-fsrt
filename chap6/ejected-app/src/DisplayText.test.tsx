import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import DisplayText from './DisplayText';
import '@testing-library/jest-dom/extend-expect';

describe("Test DisplayText", () => {
  it('renders without crashing', () => {
    const { baseElement } = render(<DisplayText />);
    expect(baseElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { baseElement } = render(<DisplayText />);
    expect(baseElement).toMatchSnapshot();
  });

  it('receives input text', () => {
    const testuser = 'testuser';
    const { getByTestId } = render(<DisplayText />);
    const input = getByTestId('user-input');
    fireEvent.change(input, { target: { value: testuser } });
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue(testuser);
  });

  it('shows welcome message', () => {
    // define values
    const testuser = 'testuser';
    const msg = `Welcome to React testing, ${testuser}`;
    // get component and deconstruct relevant method
    const { getByTestId } = render(<DisplayText />);
    // get the input
    const input = getByTestId('user-input');
    // get the button component
    const button = getByTestId('input-submit');
    // get the label component
    const label = getByTestId('final-msg');
    // fire the input change event
    fireEvent.change(input, { target: { value: testuser } });
    // fire the button event
    fireEvent.click(button)
    // test for label
    expect(label).toBeInTheDocument();
    // test for label value
    expect(label.innerHTML).toBe(msg);
  })
});
