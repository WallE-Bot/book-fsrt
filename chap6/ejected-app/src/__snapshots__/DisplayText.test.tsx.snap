// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DisplayText matches snapshot 1`] = `
<body>
  <div>
    <form>
      <label>
        Enter your name
      </label>
      <input
        data-testid="user-input"
        value=""
      />
      <button
        data-testid="input-submit"
      >
        Show Message
      </button>
      <label
        data-testid="final-msg"
      />
      <p>
        test entry
      </p>
    </form>
  </div>
</body>
`;
